pdfs/src/Pdf/Scanner.hs

```diff
@@ -1,28 +1,32 @@
 
 module Pdf.Scanner (inject, project, insertionMark, PDFBody (..), MaxSearchLength (..)) where
 
-import Data.ByteString.Char8  as LBS
-import Data.ByteString.Base64 as LBS
+import qualified Data.ByteString.Base64 as Base64
+import qualified Data.ByteString.Lazy.Char8 as LBS
+import Data.Default (Default (..))
 import Data.Hashable
 
 newtype PDFBody = PDFBody { getPDFBody :: LBS.ByteString }
+    deriving (NFData)
 
 newtype MaxSearchLength = MaxSearchLength { getMaxSearchLength :: Maybe Int }
 
-inject :: MaxSearchLength -> LBS.ByteString -> PDFBody -> Maybe PDFBody
+instance Default MaxSearchLength where
+    def = MaxSearchLength (Just 2048)
+
+inject :: MaxSearchLength -> ByteString -> PDFBody -> Maybe PDFBody
 inject
     (fromMaybe maxBound . getMaxSearchLength -> quota)
      thing
     (PDFBody text)
   =
-    let base64    = LBS.encode thing
+    let base64    = LBS.fromStrict $ Base64.encode thing
```

Comment:
I must have been very tired (or exceptionally lazy) when I reviewed the original code :man_facepalming: